build_web_compilers/lib/src/sdk_js_compile_builder.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:bazel_worker/bazel_worker.dart';
+import 'package:build/build.dart';
+import 'package:build_modules/build_modules.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import 'common.dart';
+import 'dev_compiler_builder.dart' show fixSourceMapSources;
+import 'errors.dart';
+
+const _jsSourceMapExtension = '.js.map';
+
+/// A builder which can compile the SDK to JS from kernel.
+class SdkJsCompileBuilder implements Builder {
+  /// The sdk kernel file for the current platform.
+  final String sdkKernelPath;
+
+  /// The root directory of the platform's dart SDK.
+  ///
+  /// If not provided, defaults to the directory of
+  /// [Platform.resolvedExecutable].
+  ///
+  /// On flutter this is the path to the root of the flutter_patched_sdk
+  /// directory, which contains the platform kernel files.
+  final String platformSdk;
+
+  /// The absolute path to the libraries file for the current platform.
+  ///
+  /// If not provided, defaults to "lib/libraries.json" in the sdk directory.
+  final String librariesPath;
+
+  /// The final destination for the compiled JS file.
+  final AssetId jsOutputId;
+
+  final bool soundNullSafety;
+
+  SdkJsCompileBuilder({
+    @required this.sdkKernelPath,
+    @required String outputPath,
+    @required this.soundNullSafety,
+    String librariesPath,
+    String platformSdk,
+  })  : platformSdk = platformSdk ?? sdkDir,
+        librariesPath = librariesPath ??
+            p.join(platformSdk ?? sdkDir, 'lib', 'libraries.json'),
+        buildExtensions = {
+          r'$package$': [
+            outputPath,
+            p.setExtension(outputPath, _jsSourceMapExtension),
+          ],
+        },
+        jsOutputId = AssetId('build_web_compilers', outputPath);
+
+  @override
+  final Map<String, List<String>> buildExtensions;
+
+  @override
+  Future build(BuildStep buildStep) async {
+    await _createDevCompilerModule(buildStep, platformSdk, sdkKernelPath,
+        librariesPath, jsOutputId, soundNullSafety);
+  }
+}
+
+/// Compile the sdk module with the dev compiler.
+Future<void> _createDevCompilerModule(
+  BuildStep buildStep,
+  String dartSdk,
+  String sdkKernelPath,
+  String librariesPath,
+  AssetId jsOutputId,
+  bool soundNullSafety,
+) async {
+  var scratchSpace = await buildStep.fetchResource(scratchSpaceResource);
+  var jsOutputFile = scratchSpace.fileFor(jsOutputId);
+
+  var request = WorkRequest()
+    ..arguments.addAll([
+      '--no-summarize',
```

Comment:
It's the same story with `--no-summarize` as well, it's accepted but never read. The summary is not being created by default since in the sdk build we create multiple output js module formats. We don't want copies of the same summary .dill for every output and never added a mechanism to specify the summary output path, so we just don't write the file.

Overall this seems totally confusing and it looks like we should cleanup the arguments that are accepted when compiling the sdk from dill.

Reply:
removed